lib/metasploit/framework/data_service/proxy/core.rb

```diff
@@ -15,11 +13,17 @@ module Metasploit
 module Framework
 module DataService
 class DataProxy
-  include Singleton
   include DataProxyAutoLoader
 
   attr_reader :usable
 
+  def initialize(opts = {})
```

Comment:
Changes in this file simplify initialization so that initialization doesn't have to be a separate task by removing singleton pattern.  Failing tests were caused by miss matched contexts caused indirectly by the prior initialization logic.